core/src/main/java/org/apache/brooklyn/core/location/AbstractLocation.java

```diff
@@ -718,6 +719,35 @@ public boolean removeChild(Location child) {
         return removed;
     }
 
+    public void init() {
+        super.init();
+        loadExtension();
+    }
+
+    public void rebind() {
+        super.rebind();
+        loadExtension();
+    }
+
+    private void loadExtension() {
+        try {
+            Map<String, String> extensions = getConfig(LocationConfigKeys.EXTENSIONS);
+            if (extensions != null) {
+                for (Map.Entry<String, String> extension: extensions.entrySet()) {
+                    Class<?> extensionClassType =  new ClassLoaderUtils(this, getManagementContext()).loadClass(extension.getKey());
+
+                    if (!hasExtension(extensionClassType)) {
+                        Object extensionClass = new ClassLoaderUtils(this, getManagementContext()).loadClass(extension.getValue()).newInstance();
+                        addExtension((Class)extensionClassType, extensionClass);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            LOG.error("Location extension can not be loaded {}", e);
```

Comment:
Minor: this will call `LOG.error(String msg, Throwable t)` so the `{}` has no effect. That might be what we want, to ensure the stacktrace definitely gets logged (depending what happens to the propagated exception).

If we do want just the exception's toString, would need to change it to something like:

    LOG.error("Location extension can not be loaded (rethrowing): {}", new Object[] {e});